File: example/java/config/other/Signin.java
package config.other;

public class Signin {
    private int id;
    private java.util.Map<Integer, Integer> item2countMap;
    private java.util.Map<Integer, Integer> vipitem2vipcountMap;
    private int viplevel;
    private String iconFile;
    private Signin() {
    }

    Signin(SigninBuilder b) {
        this.id = b.id;
        this.item2countMap = b.item2countMap;
        this.vipitem2vipcountMap = b.vipitem2vipcountMap;
        this.viplevel = b.viplevel;
        this.iconFile = b.iconFile;
    }

    public static Signin _create(configgen.genjava.ConfigInput input) {
        Signin self = new Signin();
        self.id = input.readInt();
        {
            int c = input.readInt();
            if (c == 0) {
                self.item2countMap = java.util.Collections.emptyMap();
            } else {
                self.item2countMap = new java.util.LinkedHashMap<>(c);
                for (; c > 0; c--) {
                    self.item2countMap.put(input.readInt(), input.readInt());
                }
            }
        }
        {
            int c = input.readInt();
            if (c == 0) {
                self.vipitem2vipcountMap = java.util.Collections.emptyMap();
            } else {
                self.vipitem2vipcountMap = new java.util.LinkedHashMap<>(c);
                for (; c > 0; c--) {
                    self.vipitem2vipcountMap.put(input.readInt(), input.readInt());
                }
            }
        }
        self.viplevel = input.readInt();
        self.iconFile = input.readStringInPool();
        return self;
    }

    /**
     * 礼包ID
     */
    public int getId() {
        return id;
    }

    /**
     * 普通奖励
     */
    public java.util.Map<Integer, Integer> getItem2countMap() {
        return item2countMap;
    }

    /**
     * vip奖励
     */
    public java.util.Map<Integer, Integer> getVipitem2vipcountMap() {
        return vipitem2vipcountMap;
    }

    /**
     * 领取vip奖励的最低等级
     */
    public int getViplevel() {
        return viplevel;
    }

    /**
     * 礼包图标
     */
    public String getIconFile() {
        return iconFile;
    }

    @Override
    public String toString() {
        return "(" + id + "," + item2countMap + "," + vipitem2vipcountMap + "," + viplevel + "," + iconFile + ")";
    }

    public static Signin get(int id) {
        config.ConfigMgr mgr = config.ConfigMgr.getMgr();
        return mgr.getOtherSignin(id);
    }

    public static java.util.Collection<Signin> all() {
        config.ConfigMgr mgr = config.ConfigMgr.getMgr();
        return mgr.allOtherSignin();
    }

    public static class _ConfigLoader implements config.ConfigLoader {

        @Override
        public void createAll(config.ConfigMgr mgr, configgen.genjava.ConfigInput input) {
            for (int c = input.readInt(); c > 0; c--) {
                Signin self = Signin._create(input);
                mgr.other_signin_All.put(self.id, self);
            }
        }

        @Override
        public void resolveAll(config.ConfigMgr mgr) {
            // no resolve
        }

    }

}

File: example/go/config/Other/Signin.go
package other

type Signin struct {
    id int //礼包ID
    item2countMap KeyedList<int, int> //普通奖励
    vipitem2vipcountMap KeyedList<int, int> //vip奖励
    viplevel int //领取vip奖励的最低等级
    iconFile string //礼包图标
    refVipitem2vipcountMap KeyedList<int, config.Other.Loot>
}

//getters
func (t *Signin) GetId() int {
    return t.id
}

func (t *Signin) GetItem2countMap() KeyedList<int, int> {
    return t.item2countMap
}

func (t *Signin) GetVipitem2vipcountMap() KeyedList<int, int> {
    return t.vipitem2vipcountMap
}

func (t *Signin) GetViplevel() int {
    return t.viplevel
}

func (t *Signin) GetIconFile() string {
    return t.iconFile
}

//ref properties
func (t *other.signin) GetRefVipitem2vipcountMap() KeyedList<int, config.Other.Loot> {
    return t.refVipitem2vipcountMap
}

var all []Signin
func GetAll() []Signin {
    return all[:len(all)]:len(all)]
}

var allMap map[int]Signin
func Get(key int) (Signin,bool){
    return allMap[key]
}
        public static Signin Get(int id)
        {
            Signin v;
            return all.TryGetValue(id, out v) ? v : null;
        }

var all []Signin
func GetAll() []Signin {
    return all[:len(all)]:len(all)]
}

var allMap map[IdViplevelKey]Signin
func Get(key IdViplevelKey) (Signin,bool){
    return allMap[key]
}
        public static Signin GetByIdViplevel(int id, int viplevel)
        {
            Signin v;
            return idViplevelMap.TryGetValue(new IdViplevelKey(id, viplevel), out v) ? v : null;
        }

        public static List<Signin> All()
        {
            return all.OrderedValues;
        }

        public static List<Signin> Filter(Predicate<Signin> predicate)
        {
            var r = new List<Signin>();
            foreach (var e in all.OrderedValues)
            {
                if (predicate(e))
                    r.Add(e);
            }
            return r;
        }

        internal static void Initialize(Config.Stream os, Config.LoadErrors errors)
        {
            all = new Config.KeyedList<int, Signin>();
            idViplevelMap = new Config.KeyedList<IdViplevelKey, Signin>();
            for (var c = os.ReadInt32(); c > 0; c--) {
                var self = _create(os);
                all.Add(self.Id, self);
                idViplevelMap.Add(new IdViplevelKey(self.Id, self.Viplevel), self);
            }
        }

        internal static void Resolve(Config.LoadErrors errors) {
            foreach (var v in All())
                v._resolve(errors);
        }

